faiss-search/scann/scann.py

```diff
@@ -66,20 +63,25 @@ def __init__(self, train_filepath: str = None,
 
     def build_advanced_index(self, vecs: 'np.ndarray'):
         """Load vectors into Scann indexers
-        1) (Optional) The first step is the partitioning, and this will be done
-        with .tree(...), this happens during training time, and at
-        query time, it will select the top partitions to the next stage
-        2) The second stage is the Scoring, this is where the distance
-        between the datapoints happen. All datapoints will be measured
-        if there was not partitioning.
-        3) (Optional) This is highly recommended if AH scroing was used.
+        1) (Optional) The first step is the partitioning, this will be done
+        with .tree(...) during training time,
+        and at query time it will select the top partitions
+        2) The second stage is the Scoring.
+            If partitioning isn't enabled it will measure the distance
+            between the query and all datapoints.
+            If partitioning is enabled it will measure only within the
+            partition to search
+        3) (Optional) This is highly recommended if AH was used.
         It will take the top k-distances and re-compute the distance.
         Then the top-k from this new measurement will be selected.
         """
         import scann
 
         _index = scann.ScannBuilder(vecs, self.training_iterations, self.distance_measure).\
             tree(self.num_leaves, self.num_leaves_to_search, self.training_sample_size).\
-            score_ah(self.dimensions_per_block, self.anisotropic_quantization_threshold).reorder(self.reordering_num_neighbors).create_pybind()
-        return _index
+            score_ah(self.dimensions_per_block, self.anisotropic_quantization_threshold).\
+            reorder(self.reordering_num_neighbors).create_pybind()
+        return self._index
 
+    def query(self, keys: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.ndarray', 'np.ndarray']:
+        neighbors, distances = self._index.search_batched(keys)
```

Comment:
since we assume that `reordering_num_neighbors`, we should add:

```python
if (self.reordering_num_neighbors < self.top_k):
    self.logger.warning('bla bla your reordering_num_neighbors should be larger or equal than ...')
```

And then, since `neighbors and distances` will have lentgh `self.reordering_num_neighbors` (to be tested), you need to select the `top_k`

```